File: algorithms/Score After Flipping Matrix/Score After Flipping Matrix.cpp
class Solution {
public:
    
    int sum(vector<vector<int>> mat){
        int score = 0;
        
        
        
        for(int i=0;i<mat.size();i++){
            int p = mat[0].size()-1;
            for(int j=0;j<mat[0].size();j++){
                score+= mat[i][j]*pow(2,p);
                p--;
            }
        }
        
        return score;
    }
    
    int matrixScore(vector<vector<int>>& grid) {
//         if(grid.size()==1){
//             return ;
//         }
        
        
        for(int i=0;i<grid.size();i++){
            if(grid[i][0]==0){
               for(int j=0;j<grid[0].size();j++){
                   if(grid[i][j]==0){
                       grid[i][j]=1;
                   }
                   else{
                        grid[i][j]=0;
                   }
               }
            }
        }
        
        
        
        
        vector<int> ind;
        
        for(int i=1;i<grid[0].size();i++){
            int ct=0;
            for(int j=0;j<grid.size();j++){
                ct+=grid[j][i];
               
            }
            if(ct<=grid.size()/2){
                ind.push_back(i);
            }
        }
        for(auto v:ind){
            cout<<v<<" ";
        }
        
        for(auto c:ind){
            for(int j=0;j<grid.size();j++){
                if(grid[j][c]==0){
                       grid[j][c]=1;
                   }
                   else{
                        grid[j][c]=0;
                   }
            }
        }
       
        
        
        
        
        return sum(grid);
        
        
        
        
        
        
       
    }

File: scripts/algorithms/S/Score After Flipping Matrix/Score After Flipping Matrix.java
// Runtime: 4 ms (Top 17.33%) | Memory: 42.9 MB (Top 9.03%)
class Solution {
    public int matrixScore(int[][] grid) {
        ArrayList<Integer> arr=new ArrayList<Integer>();
        for(int i=0;i<grid.length;i++)
        {
            boolean flip=false;
            if(grid[i][0]==0)
            {
                grid[i][0]^=1;
                flip=true;
            }
            if(flip)
            {
             for(int j=1;j<grid[0].length;j++)
            {
                grid[i][j]^=1;
            }
            }
        }
        for(int i=1;i<grid[0].length;i++)
        {
            int zero=0;
            int one=0;
            for(int j=0;j<grid.length;j++)
            {
                if(grid[j][i]==0)
                {
                    zero++;
                }
                else
                {
                    one++;
                }
            }
            if(zero>one)
            {
                arr.add(i);
            }
        }
        for(int i:arr)
        {
            for(int j=0;j<grid.length;j++)
            {
                grid[j][i]^=1;
            }
        }
        int sum=0;
        for(int i=0;i<grid.length;i++)
        {
            String num="";
            for(int j=0;j<grid[0].length;j++)
            {
                num+=String.valueOf(grid[i][j]);
            }
            sum+=Integer.valueOf(num,2);
        }
        return sum;
    }
}